Add documentation for Analysis messsage IST0134 ServiceEntryAddresses… (#9020)
istio/istio.io: content/en/docs/reference/config/analysis/ist0134/index.md

---
title: ServiceEntryAddressesRequired
layout: analysis-message
owner: istio/wg-user-experience-maintainers
test: no
---

> This message occurs when ServiceEntry with protocol TCP (or unset) protocol doesn't have addresses defined in the manifest.

Reviewer: ```suggestion
This message occurs when a ServiceEntry with the `protocol` field not set, or set to `TCP`, doesn't have `addresses` defined.
```

## Example

You will receive this message:

{{< text plain >}}

> Warning [IST0134] (ServiceEntry service-entry.default serviceentry.yaml:13) ServiceEntry addresses are required for this protocol.

Reviewer: Are `addresses` really _required_? The "How to resolve" section below makes it sound like it's not - it'll just bind all traffic to the port. This is a little confusing. @howardjohn

Reviewer: Maybe describe the unexpected behavior that the use will see if they don't define addresses.

Author: @rvennam please check this, https://github.com/istio/istio/issues/27990#issuecomment-709593864

Reviewer: Thanks. See my proposed suggestion below

Reviewer 2: It is not required, I suggest we change this to "recommended"

{{< /text >}}

> when your cluster have the following ServiceEntry with unset protocol and missing addresses:

Reviewer: ```suggestion
when your cluster has the following ServiceEntry with unset protocol and missing addresses:
```

{{< text yaml >}}
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: service-entry
  namespace: default
spec:
  hosts:
    - 'istio.io'
  exportTo:
    - "."
  ports:
    - number: 443
      name: https
  location: MESH_EXTERNAL
  resolution: DNS
{{< /text >}}

> another example of this analyzer is when you have ServiceEntry with TCP protocol and missing addresses:

Reviewer: ```suggestion
another example of this analyzer is when you have a ServiceEntry with TCP protocol and missing addresses:
```

{{< text yaml >}}
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: service-entry
  namespace: default
spec:
  hosts:
    - 'istio.io'
  exportTo:
    - "."
  ports:
    - number: 443
      name: https
      protocol: TCP
  location: MESH_EXTERNAL
  resolution: DNS
{{< /text >}}

## How to resolve

> Make sure to set `addresses` in ServiceEntry when using protocol TCP or unset protocol to avoid binding all traffic on the port define in the ServiceEntry.

Reviewer: ```suggestion
Make sure to set `addresses` in your ServiceEntry when `protocol` is not set, or set to TCP, to avoid binding all traffic on the port defined in the ServiceEntry.
```

> Make sure to set `addresses` in your ServiceEntry when `protocol` is not set, or set to TCP, to avoid binding all traffic on the port defined in the ServiceEntry.

Reviewer: ```suggestion
Make sure to set `addresses` in your ServiceEntry when `protocol` is not set, or set to TCP. If `addresses` is not set, all traffic on the port defined in the ServiceEntry is matched, regardless of the host.
```
